catroid/src/main/java/org/catrobat/catroid/ui/SpriteActivity.java

```diff
@@ -825,4 +827,11 @@ public void handlePlayButton(View view) {
 		}
 		StageActivity.handlePlayButton(ProjectManager.getInstance(), this);
 	}
+
+	void notifyDataSetChanged() {
+		ScriptFragment parentFragment = (ScriptFragment) getSupportFragmentManager().findFragmentByTag(ScriptFragment.TAG);
```

Comment:
you can use `getCurrentFragment()` here.

Reply:
and maybe don't call it `parentFragment` because that would IMO imply that it's the parent of the current activity, which it isn't. perhaps something like `scriptFragment` because it has to be the `ScriptFragment` anyway. 

which besides, makes me wonder if you should really call the function `notifyDataSetChanged()` because it's more specific than that. it just notifies the fragment if it is the `ScriptFragment`. so maybe use something like `refeshScriptFragment()`.